src/history/hs.c

```diff
@@ -79,22 +118,21 @@ __wt_hs_config(WT_SESSION_IMPL *session, const char **cfg)
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    WT_ERR(__hs_start_internal_setup_session(session, &temp_setup_session));
+    release_temp_session = true;
```

Comment:
Is the boolean tracking the session necessary? It looks like we could just check against a NULL `temp_setup_session` pointer, and it would do the same thing?